ci: pnpm audit fails permanently (npm audit API retired, HTTP 410)

## Problem

The `Security audit` CI job (`pnpm audit --audit-level=high` in `.github/workflows/main.yml`) now fails on every PR/push with:

\`\`\`
ERR_PNPM_AUDIT_BAD_RESPONSE  The audit endpoint (at https://registry.npmjs.org/-/npm/v1/security/audits/quick) responded with 410: {"error":"This endpoint is being retired. Use the bulk advisory endpoint instead. See the following docs for more info: https://api-docs.npmjs.com/#tag/Audit"}. Fallback endpoint (at https://registry.npmjs.org/-/npm/v1/security/audits) responded with 410: ...
\`\`\`

Confirmed this is not pnpm-version-specific: reproduced with both the pinned `pnpm@10.32.1` and the latest `pnpm@11.13.0`. The underlying npm registry audit endpoints (`/quick` and the fallback) have both been retired in favor of a "bulk advisory endpoint" that pnpm does not yet call.

Since `develop` has no branch protection, this doesn't block merges today, but the check is effectively dead — it will show red on every PR indefinitely until either pnpm ships support for the new endpoint, or the workflow switches to a different mechanism.

## Options considered (not yet decided)

- Wait for pnpm to add support for the new bulk advisory endpoint, and bump once available.
- Rely on GitHub's own Dependabot alerts / CodeQL (already running in this repo) instead of `pnpm audit`, dropping the CI step.
- Swap in a different scanning tool that doesn't depend on the retired npm endpoint (e.g. `osv-scanner`).

## Completion criteria
- `Security audit` (or its replacement) reflects real vulnerability status again, not a permanent infrastructure-level failure.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -13,18 +13,20 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
-  audit:
-    name: Security audit
+  # Replaces the former `pnpm audit` job: the npm registry retired the audit
+  # endpoints it relied on (HTTP 410, see #695). Dependency review checks the
+  # dependency diff of each PR against the GitHub Advisory Database; existing
+  # dependencies are covered by Dependabot alerts, weekly CodeQL, and pnpm's
+  # minimumReleaseAge.
+  dependency-review:
+    name: Dependency review
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v6
-      - uses: pnpm/action-setup@v4
-      - uses: actions/setup-node@v6
+      - uses: actions/dependency-review-action@v4
         with:
-          node-version: '24.x'
-          cache: pnpm
-      - run: pnpm install --frozen-lockfile
-      - run: pnpm audit --audit-level=high
+          fail-on-severity: high
 
   build:
     name: Build, lint, and test on Node ${{ matrix.node }}

--- a/.github/workflows/visual-regression.yml
+++ b/.github/workflows/visual-regression.yml
@@ -38,8 +38,24 @@ jobs:
 
       - run: pnpm install --frozen-lockfile
 
+      - name: Get Playwright version
+        id: playwright-version
+        run: echo "version=$(pnpm ls playwright --depth=0 --parseable --long | grep -oP 'playwright@\K[0-9.]+' | head -1)" >> $GITHUB_OUTPUT
+
+      - name: Cache Playwright browsers
+        id: playwright-cache
+        uses: actions/cache@v5
+        with:
+          path: ~/.cache/ms-playwright
+          key: playwright-${{ runner.os }}-${{ steps.playwright-version.outputs.version }}
+
       - name: Install Playwright browsers
+        if: steps.playwright-cache.outputs.cache-hit != 'true'
         run: pnpm exec playwright install --with-deps chromium
+
+      - name: Install Playwright system dependencies only (cache hit)
+        if: steps.playwright-cache.outputs.cache-hit == 'true'
+        run: pnpm exec playwright install-deps chromium
 
       # On develop: generate and cache reference screenshots
       - name: Restore reference screenshots (develop)

